### ISSUE TITLE
stream writer should have options to overwrite vs append

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#142]
Some formats like json are broken when appended.

CSV can be appended but only if the header fields are not repeated

I think 3 options:
1) overwrite
2) append

**Comment by jaypha** [comment]
Keep in mind that overwrite does not make sense for all streams. For example, you cannot overwrite to 'php://output'.

**Comment by brendanheywood** [comment]
True, but even in that case it still might affect whether or not we start with a header row with csv for example
